fam_copy data items across memory server

faizan-barmawer opened the pull request:
fam_copy between data item from different memory server to be performed from destination memory server.

faizan-barmawer commented:
Updated the PR with all the changes.
- Removed commented out code.
- added all the changes required from memory server side.

mashoodk commented on src/fam-api/fam_ops_libfabric.cpp:
```diff
@@ -177,6 +180,10 @@ int Fam_Ops_Libfabric::initialize() {
                                 message.str().c_str());
             }
 
+            // Save memery server address in memServerAddrs map
```
```suggestion
            // Save memory server address in memServerAddrs map
```

mashoodk reviewed (changes requested):
very minor comment.

faizan-barmawer commented:
Pushed a commit with additional changes:
    - Added a pthread_rw_lock around FiMemsrvMap, required on server side, since memory-server is a multi-threaded application
    - Moved FiMemsrvMap from memory_registrion_libfabric.h to fam_ops_libfabric
    - Added a test case

Regards,
Faizan
